src/libfetchers/git.cc

```diff
@@ -24,6 +27,212 @@ static std::string readHead(const Path & path)
     return chomp(runProgram("git", true, { "-C", path, "rev-parse", "--abbrev-ref", "HEAD" }));
 }
 
+static string getRootTree(const Path & path, const string & gitrev) {
+    auto commitOutput = runProgram("git", true, { "-C", path, "cat-file", "-p", gitrev });
+    auto lines = tokenizeString<std::vector<string>>(commitOutput, "\n");
+    auto words = tokenizeString<std::vector<string>>(lines[0]);
+    auto rootTree = words[1];
+    return rootTree;
+}
+
+enum gitType {
+  blob,
+  tree,
+  commit
+};
+
+string getBlob(const Path & path, const string & gitrev, const string & blobhash) {
```

Comment:
Forgive my ignorance, but is there a reason this isn't `static` (given that `getRootTree` is)?

Reply:
Probably because I suck at C++. @cleverca22 do you have a better answer? 😅